airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/DateTimeConverter.java

```diff
@@ -145,10 +145,10 @@ public static String convertToTime(final Object time) {
       return localTime.format(TIME_FORMATTER);
     } else if (time instanceof java.time.Duration) {
       long value = ((Duration) time).toNanos();
-      if (value >= 0 && value <= TimeUnit.DAYS.toNanos(1)) {
+      if (value >= 0 && value < TimeUnit.DAYS.toNanos(1)) {
         return LocalTime.ofNanoOfDay(value).format(TIME_FORMATTER);
       } else {
-        final long updatedValue = 0 > value ? Math.abs(value) : TimeUnit.DAYS.toNanos(1);
+        final long updatedValue = 0 > value ? Math.abs(value) : 86399999999999L;
```

Comment:
this would be more clear, assuming I understood correctly
```suggestion
        final long updatedValue = 0 > value ? Math.abs(value) : TimeUnit.DAYS.toNanos(1) - 1;
```

or maybe just this? I.e. convert negative values to positive, and then cap all values to less than 23:59 - this is slightly different from the current behavior, where `-864000...0000` would be converted to `864000...0000` rather than being reduced to 86399999999999)
```suggestion
        final long updatedValue = Math.min(Math.abs(value), TimeUnit.DAYS.toNanos(1) - 1);
```

Reply:
done